ci: 🎡 new Workflow for lint and type-check on PRs

IdrisGit opened the pull request:
### Description
Added a new workflow file that runs whenever a PR is opened, synchronized, or reopened against the main branch.

This provides a CI check to make sure the linting is correct for the project.

✅ Closes: #264

growupanand commented on .github/workflows/open-pr.yml:
```diff
@@ -0,0 +1,40 @@
+        with:
+          node-version: "lts/*"
+          cache: "pnpm"
+      - name: Install dependencies
```
Why do we need to install all dependencies again?

IdrisGit replied:
`actions/setup-node` have built in support for `actions/cache`, when we install dependencies in the first job i.e `lint`, it will cache the dependencies for the subsequent jobs, in the `type-check` it will fetch dependencies from cache instead of downloading them, or it will download if there is a cache miss.

growupanand commented:
Can you please create a test PR in your branch to test this workflow before merging?

IdrisGit commented:
> Can you please create a test PR in your branch to test this workflow before merging?

The workflow had an issue, specifically with the ppm version 9 bugs. I have updated the PR, and it's working now. In fact, it is successfully detecting check errors in some files. You can check the action run here: [PR Workflow Run](https://github.com/IdrisGit/ConvoForm/actions/runs/11484261468).

I have also added `concurrency` in the workflow, which will prevent multiple action runs of the same PR if changes are pushed while one is in progress and will only run the latest push. This should help with organization and costs.

growupanand commented:
> > Can you please create a test PR in your branch to test this workflow before merging?
> 
> The workflow had an issue, specifically with the ppm version 9 bugs. I have updated the PR, and it's working now. In fact, it is successfully detecting check errors in some files. You can check the action run here: [PR Workflow Run](https://github.com/IdrisGit/ConvoForm/actions/runs/11484261468).
> 
> I have also added `concurrency` in the workflow, which will prevent multiple action runs of the same PR if changes are pushed while one is in progress and will only run the latest push. This should help with organization and costs.

Sorry for the delay, Merged the PR.
